analysis_options.yaml

```diff
@@ -1,42 +1,6 @@
-include: package:pedantic/analysis_options.yaml
+include: package:pedantic/analysis_options.1.9.0.yaml
 analyzer:
   errors:
-    unused_element: error
-    unused_import: error
-    unused_local_variable: error
-    dead_code: error
-linter:
-  rules:
-    # Errors
-    - avoid_empty_else
-    - control_flow_in_finally
-    - empty_statements
-    - hash_and_equals
-    - implementation_imports
-    - test_types_in_equals
-    - throw_in_finally
-    - unrelated_type_equality_checks
-    - valid_regexps
-
-    # Style
-    - avoid_init_to_null
-    - avoid_private_typedef_functions
-    - avoid_renaming_method_parameters
-    - avoid_return_types_on_setters
-    - await_only_futures
-    - camel_case_types
-    - directives_ordering
-    - empty_catches
-    - empty_constructor_bodies
-    - library_names
-    - library_prefixes
-    - non_constant_identifier_names
-    - only_throw_errors
-    - prefer_equal_for_default_values
-    - prefer_final_fields
-    - prefer_generic_function_type_aliases
-    - prefer_is_not_empty
-    - slash_for_doc_comments
-    - type_init_formals
-    - unnecessary_const
-    - unnecessary_new
+    annotate_overrides: ignore
+    prefer_single_quotes: ignore
+    use_function_type_syntax_for_parameters: ignore
```

Comment:
We should be following this lint.

Some of the removed ones that aren't in `pedantic` yet are also useful. We rarely violate them on the team, but I don't see any reason for something like `camel_case_types` or `library_prefixes` to need to wait for code review if we do get a contribution from outside the team.

Reply:
As you may  be aware, I disagree with following that lint. It's not part of the Dart style guide, and I actually prefer the inline notation for function arguments, both for reading and writing.
These are the three exceptions that I apply because I consistently write code that does not follow them.

I'm using pedantic+modifications because it's convenient (there are good things in there too, just not all of it), not because I think it's what we actually want as the Dart recommended lint set outside of Google3.
(Sure, I can add camel_case_types, but that has never been an issue.)

Reply:
How is it possibly a good thing to have every random person who touches a codebase changing the lints to fit their personal preference?

Reply:
> It's not part of the Dart style guide, and I actually prefer the inline notation for function arguments, both for reading and writing.

https://dart.dev/guides/language/effective-dart/design#consider-using-function-type-syntax-for-parameters

It is a "Consider" it think for largely historical reasons. In future revisions it's likely to become stronger. The fact that there _are_ 2 syntaxes for this will hopefully be fixed in Dart 3 or some 2.x language version - ideally we have 1 way to do things instead of 2 competing ways that require constant negotiation or inconsistent codebases.

> These are the three exceptions that I apply because I consistently write code that does not follow them.

This goes against the flow of the rest of Google Dart authors, and against the general philosophy of Readability in Google.

> Sure, I can add camel_case_types, but that has never been an issue.

It has never been a _big_ issue. It has come up in code reviews in the past and the intention of putting it in a lint is to save time, and save from re-negotiating it with new contributors. The reason `camel_case_types` isn't in `pedantic` is because that would require consistent enforcement, and in internal code I think we have some exceptions. We won't have exceptions in _this_ package which is why the lint is enabled.


I know that we have different approaches to linting in different places. I don't think having us each try to assert our own style into shared packages is the right thing to do, neither is introducing config changes into other PRs.